tests/metadata_stores/test_lancedb.py

```diff
@@ -0,0 +1,286 @@
+"""LanceDB-specific tests.
+
+Most LanceDB store functionality is tested via parametrized tests in StoreCases.
+This module tests LanceDB-specific features:
+- Connection property enforcement
+- Path type detection (local vs remote)
+- URI handling (mkdir behavior for local vs remote)
+- Credential sanitization in display()
+- Object storage integration (S3) with storage_options
+"""
+
+from __future__ import annotations
+
+from pathlib import Path
+from unittest.mock import MagicMock, Mock
+
+import polars as pl
+import pytest
+
+from metaxy import FeatureKey
+from metaxy._utils import collect_to_polars
+from metaxy.config import MetaxyConfig
+from metaxy.metadata_store import StoreNotOpenError
+from metaxy.metadata_store.lancedb import LanceDBMetadataStore
+
+
+@pytest.fixture(autouse=True)
+def _lancedb_project(config):
+    """Ensure LanceDB tests run with the same project as test features."""
+    original_config = MetaxyConfig.get()
+    project_config = config.model_copy(update={"project": "test_stores"})
+    MetaxyConfig.set(project_config)
+    yield
+    MetaxyConfig.set(original_config)
+
+
+def test_lancedb_table_naming(tmp_path, test_graph, test_features) -> None:
+    """Test that feature keys are converted to table names correctly."""
+    database_path = tmp_path / "lancedb"
+    feature_cls = test_features["UpstreamFeatureA"]
+
+    with LanceDBMetadataStore(database_path) as store:
+        metadata = pl.DataFrame(
+            {
+                "sample_uid": [1],
+                "metaxy_provenance_by_field": [{"frames": "h1", "audio": "h1"}],
+            }
+        )
+        store.write_metadata(feature_cls, metadata)
+
+        # Verify table was created (uses optimized _table_exists internally)
+        assert store.has_feature(feature_cls, check_fallback=False)
+
+
+def test_lancedb_conn_property_enforcement(tmp_path, test_graph, test_features) -> None:
+    """Test that conn property enforces store is open."""
+    database_path = tmp_path / "lancedb"
+    store = LanceDBMetadataStore(database_path)
+
+    # Should raise when accessing conn while closed
+    with pytest.raises(StoreNotOpenError, match="LanceDB connection is not open"):
+        _ = store.conn
+
+    with store:
+        conn = store.conn
+        assert conn is not None
+
+
+def test_lancedb_is_local_path() -> None:
```

Comment:
should just have this shared with deltalake

Reply:
we should also have it shared with PG (for the display/sanitize)